editor/scene_tree_editor.cpp

```diff
@@ -839,6 +839,42 @@ void SceneTreeEditor::_rename_node(ObjectID p_node, const String &p_name) {
 	item->set_text(0, p_name);
 }
 
+void SceneTreeEditor::_renaming(String p_text) {
+	TreeItem *which = tree->get_edited();
+
+	ERR_FAIL_COND(!which);
+	NodePath path = which->get_metadata(0);
+	Node *edited_node = get_node(path);
+	ERR_FAIL_COND(!edited_node);
+
+	bool valid_node_name = true;
+
+	// Empty node names are not allowed.
+	if (p_text.strip_edges().is_empty() || p_text != p_text.validate_node_name()) {
+		valid_node_name = false;
+
+	// Unique Node with the same name already exists in owner.
+	} else if (edited_node->is_unique_name_in_owner()) {
+
+		String potato = UNIQUE_NODE_PREFIX + p_text.validate_node_name().strip_edges();
+		Node* found_unique_node = get_tree()->get_edited_scene_root()->get_node_or_null(potato);
+		if (found_unique_node != nullptr && found_unique_node != edited_node) {
+			valid_node_name = false;
+		}
+	}
+
+	if (valid_node_name) {
+		tree->get_text_editor()->remove_theme_style_override("focus");
+	} else {
+		Ref<StyleBox> stylebox = get_theme_stylebox("error", "LineEdit");
+
+		tree->get_text_editor()->add_theme_style_override("focus", stylebox);
+		//tree->get_text_editor()->set_modulate(Color(1, 0.47, 0.42));
```

Comment:
Remember to remove commented out code before this can be merged.

Reply:
No worries! I'll clean up properly when I manage to fix the aforementioned bug.